packages/bigquery/src/job.js

```diff
@@ -266,14 +267,41 @@ Job.prototype.cancel = function(callback) {
  * });
  */
 Job.prototype.getQueryResults = function(options, callback) {
+  var self = this;
+
   if (is.fn(options)) {
     callback = options;
-    options = {};
+    options = null;
   }
 
-  options = options || {};
-  options.job = this;
-  this.bigQuery.query(options, callback);
+  this.bigQuery.request({
+    uri: '/queries/' + this.id,
+    qs: options
+  }, function(err, resp) {
+    if (err) {
+      callback(err, null, null, resp);
+      return;
+    }
+
+    var rows = [];
+
+    if (resp.schema && resp.rows) {
+      rows = self.bigQuery.mergeSchemaWithRows_(resp.schema, resp.rows);
+    }
+
+    var nextQuery = null;
+    if (resp.jobComplete === false) {
```

Comment:
I can't tell what's going on here. Is this polling?

Reply:
@tswast yep! We recursively call any rpcs that require pagination until we have received all the data. We generally stop when we do not receive a pageToken, however in this case we also check to see if the job is completed or not.

Reply:
This concerns me. The result set for a query or table data listing can be very large (hundreds of megabytes to gigabytes, or even petabytes). This implementation means someone could not use this library to process such large result sets that do not fit memory or have too many pages that they blow the stack.

Reply:
Our pagination methods also allow `autoPaginate: false`, where a user will be able to page through the responses at the rate they wish. There is an example here: https://googlecloudplatform.github.io/google-cloud-node/#/docs/bigquery/0.9.6/bigquery/job?method=getQueryResults

```js
function manualPaginationCallback(err, rows, nextQuery, apiResponse) {
  if (nextQuery) {
    // More results exist.
    job.getQueryResults(nextQuery, manualPaginationCallback);
  }
}

job.getQueryResults({
  autoPaginate: false
}, manualPaginationCallback);
```

Reply:
Is it possible to pass in `autoPaginate` false when the job is created / query is started? It's a bit dangerous to automatically loop through the results, especially if `allowLargeResults` is set in the query job.

Reply:
For a definitive answer, I'll defer to @callmehiphop. The way the API is currently (without this PR), the user can pass `autoPaginate: false` to `bigquery.query(...)`, which would work the same was as the example in my last post.

Manually paginating was actually the default behavior when this library started. Across many APIs, we share the concern that the incoming bytes can be too large. I believe the rationale to default autoPaginate to true was the motto we hear throughout discussions like these, something like: "make the common case easy and the advanced case possible".

Reply:
It is, however to re-use the same job the user would need to use the `startQuery` method to cache the job object.